blob_io_test.go

```diff
@@ -0,0 +1,251 @@
+// Copyright (C) 2022 Yasuhiro Matsumoto <mattn.jp@gmail.com>.
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file.
+
+//go:build cgo
+// +build cgo
+
+package sqlite3
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"io"
+	"testing"
+)
+
+// Verify interface implementations
+var _ io.Reader = &SQLiteBlob{}
+var _ io.Writer = &SQLiteBlob{}
+var _ io.Seeker = &SQLiteBlob{}
+var _ io.Closer = &SQLiteBlob{}
+
+func blobTestData(dbname string, rowid int64, blob []byte) (*sql.DB, *SQLiteConn, error) {
+	db, err := sql.Open("sqlite3", "file:"+dbname+"?mode=memory&cache=shared")
+	if err != nil {
+		return nil, nil, err
+	}
+
+	conn, err := db.Conn(context.Background())
+	if err != nil {
+		return nil, nil, err
+	}
+	defer conn.Close()
+
+	var driverConn *SQLiteConn
+	err = conn.Raw(func(conn interface{}) error {
+		driverConn = conn.(*SQLiteConn)
```

Comment:
You should not do this. It is a violation of `conn.Raw` to reference the connection outside the callback.
For your tests to work properly, please change `blobTestData` to take a callback `func(*testing.T, *SQLiteConn)`.